Tools/Linter/runClangTidy.sh

```diff
@@ -0,0 +1,157 @@
+#
+# Note: this script is only tested on Linux
+
+echo "============================================="
+echo
+echo "This script is a developer's tool to perform the"
+echo "checks done by the clang-tidy CI test locally"
+echo "_____________________________________________"
+
+# Check source dir
+REPO_DIR=$(cd $(dirname ${BASH_SOURCE})/../../ && pwd)
+echo
+echo "Your current source directory is: ${REPO_DIR}"
+echo "_____________________________________________"
+
+# Set number of jobs to use for compilation
+PARALLEL="${WARPX_TOOLS_LINTER_PARALLEL:-4}"
+echo
+echo "${PARALLEL} jobs will be used for compilation."
+echo "This can be overridden by setting the environment"
+echo "variable WARPX_TOOLS_LINTER_PARALLEL, e.g.: "
+echo
+echo "$ export WARPX_TOOLS_LINTER_PARALLEL=8"
+echo "$ ./Tools/Linter/runClangTidy.sh"
+echo "_____________________________________________"
+
+# Check clang version
+export CC="${CLANG:-"clang"}"
+export CXX="${CLANGXX:-"clang++"}"
+export CTIDY="${CLANGTIDY:-"clang-tidy"}"
+echo
+echo "The following versions of the clang compiler and"
+echo "of the clang-tidy linter will be used:"
+echo
+echo "clang version:"
+which ${CC}
+${CC} --version
+echo
+echo "clang++ version:"
+which ${CXX}
+${CXX} --version
+echo
+echo "clang-tidy version:"
+which ${CTIDY}
+${CTIDY} --version
+echo
+echo "This can be overridden by setting the environment"
+echo "variables CLANG, CLANGXX, and CLANGTIDY e.g.: "
+echo "$ export CLANG=clang-15"
+echo "$ export CLANGXX=clang++-15"
+echo "$ export CTIDCLANGTIDYY=clang-tidy-15"
+echo "$ ./Tools/Linter/runClangTidy.sh"
+echo
+echo "*****************************************************"
+echo "* Warning: clang v15 is currently used in CI tests. *"
+echo "* It is therefore recommended to use this version.  *"
+echo "* Otherwise, a newer version may find issues not    *"
+echo "* currently covered by CI tests (checks are opt-in) *"
+echo "* while older versions may not find all the issues. *"
+echo "*****************************************************"
+echo "_____________________________________________"
+
+# Install BLAS++
+echo
+echo "Installing BLAS++"
+echo
+if [ -d ${REPO_DIR}/src/blaspp ]
+then
+  cd ${REPO_DIR}/src/blaspp
+  git fetch --prune
+  git checkout master
+  git pull
+  cd -
+else
+  git clone https://github.com/icl-utk-edu/blaspp.git ${REPO_DIR}/src/blaspp
+fi
+rm -rf ${REPO_DIR}/src/blaspp-build
+cmake -S ${REPO_DIR}/src/blaspp -B ${REPO_DIR}/src/blaspp-build -Duse_openmp=ON -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=${REPO_DIR}/blaspp-install
+cmake --build ${REPO_DIR}/src/blaspp-build --target install --parallel ${PARALLEL}
+rm -rf ${REPO_DIR}/src/blaspp-build
+export CMAKE_PREFIX_PATH=${REPO_DIR}/blaspp-install:${CMAKE_PREFIX_PATH}
+echo "_____________________________________________"
+
+# Install LAPACK++
+echo
+echo "Installing LAPACK++"
+echo
+if [ -d ${REPO_DIR}/src/lapackpp ]
+then
+  cd ${REPO_DIR}/src/lapackpp
+  git fetch --prune
+  git checkout master
+  git pull
```

Comment:
I was thinking users would run this script in their current developer environment, e.g., https://warpx.readthedocs.io/en/latest/install/dependencies.html#conda-linux-macos-windows, and do not install dependencies again.

Reply:
You are right. I have removed it. Both the script and the documentation now recommends to ensure that all the dependencies of WarpX are met.